ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -7324,6 +7324,158 @@ bool GDALGeoPackageDataset::DeleteVectorOrRasterLayer(const char *pszLayerName)
+        return true;
+    }
+
+    char *pszSQL =
+        sqlite3_mprintf("SELECT 1 FROM gpkg_contents WHERE "
+                        "lower(table_name) = lower('%q') "
+                        "AND data_type IN ('tiles', '2d-gridded-coverage')",
+                        pszLayerName);
+    const bool bIsRasterTable = SQLGetInteger(hDB, pszSQL, nullptr) == 1;
+    sqlite3_free(pszSQL);
+
+    if (bIsRasterTable)
+    {
+        return RenameRasterLayer(pszLayerName, pszNewLayerName);
+    }
+
+    return false;
+}
+
+bool GDALGeoPackageDataset::RenameRasterLayer(const char *pszLayerName,
+                                              const char *pszNewLayerName)
+{
+    std::string osSQL;
+
+    char *pszSQL = sqlite3_mprintf(
+        "SELECT 1 FROM sqlite_master WHERE lower(name) = lower('%q') "
+        "AND type IN ('table', 'view')",
+        pszNewLayerName);
+    const bool bAlreadyExists = SQLGetInteger(GetDB(), pszSQL, nullptr) == 1;
+    sqlite3_free(pszSQL);
+    if (bAlreadyExists)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Table %s already exists",
+                 pszNewLayerName);
+        return false;
+    }
+
+    // Temporary remove foreign key checks
+    const GPKGTemporaryForeignKeyCheckDisabler
+        oGPKGTemporaryForeignKeyCheckDisabler(this);
+
+    if (SoftStartTransaction() != OGRERR_NONE)
+    {
+        return false;
+    }
+
+    osSQL = sqlite3_mprintf("UPDATE gpkg_contents SET table_name = '%q' WHERE "
+                            "lower(table_name) = lower('%q');",
+                            pszNewLayerName, pszLayerName);
+
+    osSQL += sqlite3_mprintf("UPDATE gpkg_contents SET identifier = '%q' WHERE "
+                             "lower(identifier) = lower('%q');",
+                             pszNewLayerName, pszLayerName);
+
+    osSQL +=
+        sqlite3_mprintf("UPDATE gpkg_tile_matrix SET table_name = '%q' WHERE "
+                        "lower(table_name) = lower('%q');",
+                        pszNewLayerName, pszLayerName);
+
+    osSQL += sqlite3_mprintf(
+        "UPDATE gpkg_tile_matrix_set SET table_name = '%q' WHERE "
+        "lower(table_name) = lower('%q');",
+        pszNewLayerName, pszLayerName);
+
+    if (HasExtensionsTable())
+    {
+        osSQL += sqlite3_mprintf(
+            "UPDATE gpkg_extensions SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+    }
+
+    if (HasMetadataTables())
+    {
+        osSQL += sqlite3_mprintf("UPDATE gpkg_metadata SET id = '%q' WHERE "
+                                 "lower(id) = lower('%q');",
+                                 pszNewLayerName, pszLayerName);
+    }
+
+    if (HasDataColumnsTable())
+    {
+        osSQL += sqlite3_mprintf(
+            "UPDATE gpkg_data_columns SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+    }
+
```

Comment:
if HasGriddedCoverageAncillaryTable(), you should also update the tile_matrix_set_name column of gpkg_2d_gridded_coverage_ancillary, and  tpudt_name of gpkg_2d_gridded_tile_ancillary

Reply:
done